zilliqa/src/consensus.rs

```diff
@@ -513,6 +514,17 @@ impl Consensus {
             return Ok(None);
         }
 
+        if block.gas_limit() > self.config.consensus.eth_block_gas_limit
+            || block.gas_used() > block.gas_limit()
+        {
+            bail!(
+                "Block gas used/limit check failed. Used: {}, Limit: {}, config limit: {}",
+                block.gas_used(),
+                block.gas_limit(),
+                self.config.consensus.eth_block_gas_limit
+            );
```

Comment:
I think you want to log and return `Ok(None)` here - We haven't malfunctioned - Just someone has sent us an invalid block.

Reply:
And if you agree, the same point for the rest of these `bail!`s.